tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,363 @@
+                                           hints={
+                                               'datetimeformattz': datetimeformattz,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            expect_pandas_failure = not self.has_pandas() and uses_pandas
+            try:
+                csv_text = self.unload_column_to_string(column_name='timestamptz',
+                                                        records_format=records_format)
+                self.assertIn(csv_text, [create_sample(datetimeformattz) + "\n",
+                                         create_sample(datetimeformattz).
+                                         replace('-00', '+00') + "\n",
+                                         create_sample(datetimeformattz).replace('-00', '') +
+                                         ".000000\n",
+                                         # Pandas doesn't truncate fractional seconds in the
+                                         # same way other tools do.
+                                         create_sample(datetimeformattz) + ".000000\n",
+                                         # It's valid to not include an offset with no impact.
+                                         create_sample(datetimeformattz).replace('-00', '') +
+                                         '.000000+0000\n',
+                                         # An example of having an overly simplistic
+                                         # implementation of to_csv_options is that seconds get
+                                         # appended even for time formats which don't include
+                                         # it.
+                                         #
+                                         # https://github.com/bluelabsio/records-mover/issues/142
+                                         create_sample(datetimeformattz) +
+                                         f":{SAMPLE_SECOND:02d}.000000\n"
+                                         ],
+                              f"from datetimeformattz {datetimeformattz} and "
+                              f"addl_hints {addl_hints}")
+            except ModuleNotFoundError as e:
+                if 'pandas' in str(e) and expect_pandas_failure:
+                    # as expected
+                    continue
+                else:
+                    raise
+            except NotImplementedError as e:
+                if expect_am_failure and 'AM' in str(e):
+                    # as expected
+                    continue
+                else:
+                    raise
+            self.assertFalse(expect_pandas_failure)
+            self.assertFalse(expect_am_failure)
+
+    def test_unload_timeonly(self) -> None:
```

Comment:
C901 'RecordsUnloadDatetimeIntegrationTest.test_unload_timeonly' is too complex (17)

Reply:
I haven't tried much further to refactor these to reduce complexity, and quality is failing as a result.  Any suggestions?  I'm inclined to bump the numbers and merge and either declare this to be the best trade-off for now or make this a problem for the future.

Reply:
Not immediately, so I think bumping is fine.

The one thing that came to mind-- but I don't know the right abstraction off the top of my head-- is that fundamentally stuff like "use_pandas" comes down to some facts about support that the database and/or records-mover have for certain functionality or behaviors. Perhaps there are some decisions that can be pushed into standard interfaces in drivers instead-- that could reduce some of the branching.

The most straightforward idea is something like `will_use_pandas_for_unload`. That seems a little bit like exposing implementation details (and making it the problem of the end user, such as this test), rather than exposing the actual constraints. But maybe it's still better than baking the logic into the tests?